packages/hash/api/src/graphql/resolvers/pages/accountPages.ts

```diff
@@ -7,10 +7,12 @@ export const accountPages: Resolver<
   {},
   GraphQLContext,
   QueryAccountPagesArgs
-> = async (_, { accountId }, { dataSources }) => {
+> = async (_, { accountId, archived }, { dataSources }) => {
   const pages = await Page.getAllPagesInAccount(dataSources.db, {
     accountId,
   });
 
-  return pages.map((page) => page.toGQLUnknownEntity());
+  return pages
+    .filter((page) => !!page.properties.archived === archived)
```

Comment:
I'm not sure this filtering logic is sufficient when considering nested pages. I think the expected behaviour when you delete a top-level page should be that all its children are archived as well. The current behaviour looks like this:

https://user-images.githubusercontent.com/42802102/175516660-38d823ca-7e9a-48f0-8260-0328070739b9.mov

Just to clarify I don't think that archiving a page should set the `properties.archived` properties to `true` for all its children, as a page can have more than one parent page.

Therefore I think the filtering logic needs to filter:
- pages that are archived (where `properties.archived === true`), and
- pages which aren't archived, have one or more parent, but shouldn't appear in the page tree because their parents (or grandparents, etc) are archived

Reply:
I don't think a page can have multiple parents – any existing parent is deleted when a new one is set. See [here](https://github.com/hashintel/hash/blob/f95407bc1e251a80e965a9d2512f05d79869b5fb/packages/hash/api/src/model/page.model.ts#L278).

Therefore it should be sufficient to:
1. Find all the children of the page that's been archived
2. For each, set them to archived, and goto 1

Reply:
Addressed in 577d675a637c4fd087f1582ddf7c29e65ddba753

Reply:
Thanks Luis, this does account for sub-pages. However you've implemented a GQL mutation that updates the properties of all the sub-pages of a page, rather than just updating the properties of the page the user is trying to archive (as you had it before) and updating the _filtering logic_ to account for un-archived pages that may have parents that are archived (as I suggested in https://github.com/hashintel/hash/pull/690#discussion_r905940745). I think this approach would be better because:
- it removes the need for the `archivePage` GQL mutation you've implemented (you can just re-use `updateProperties` to update the properties of the page entity)
- results in less updated entities in the database (only the page that the user is archiving needs to be updated, instead of needing to update all the children)
- makes it easier to implement un-archiving a page (you can just re-use `updateProperties` to update the properties of the page you're un-archiving, rather than having to update all the child pages as well)

Reply:
We discussed this – I think since this works and may be impacted by the Graph API anyway, we should get it as is and worry about changing it if/when we introduce un-archiving, or make changes to use the Graph API